api4/plugin_test.go

```diff
@@ -1371,6 +1371,93 @@ func TestInstallMarketplacePlugin(t *testing.T) {
 		require.Nil(t, appErr)
 	})
 
+	t.Run("verify EnterprisePlugins is false for TE", func(t *testing.T) {
+		testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+			licenseType, ok := req.URL.Query()["enterprise_plugins"]
+			require.True(t, ok)
+			require.Len(t, licenseType, 1)
+			require.Equal(t, "false", licenseType[0])
+
+			res.WriteHeader(http.StatusOK)
+			json, err := json.Marshal([]*model.MarketplacePlugin{samplePlugins[0]})
+			require.NoError(t, err)
+			res.Write(json)
+		}))
+		defer func() { testServer.Close() }()
+
+		th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.PluginSettings.EnableMarketplace = true
+			*cfg.PluginSettings.EnableRemoteMarketplace = true
+			*cfg.PluginSettings.MarketplaceUrl = testServer.URL
+		})
+
+		// The content of the request is irrelevant. This test only cares about enterprise_plugins.
+		pRequest := &model.InstallMarketplacePluginRequest{}
+		manifest, resp := th.SystemAdminClient.InstallMarketplacePlugin(pRequest)
+		CheckInternalErrorStatus(t, resp)
+		require.Nil(t, manifest)
+	})
+
+	t.Run("verify EnterprisePlugins is false for E10", func(t *testing.T) {
+		testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+			licenseType, ok := req.URL.Query()["enterprise_plugins"]
+			require.True(t, ok)
+			require.Len(t, licenseType, 1)
+			require.Equal(t, "false", licenseType[0])
+
+			res.WriteHeader(http.StatusOK)
+			json, err := json.Marshal([]*model.MarketplacePlugin{})
+			require.NoError(t, err)
+			res.Write(json)
+		}))
+		defer func() { testServer.Close() }()
+
+		th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.PluginSettings.EnableMarketplace = true
+			*cfg.PluginSettings.EnableRemoteMarketplace = true
+			*cfg.PluginSettings.MarketplaceUrl = testServer.URL
+		})
+
+		l := model.NewTestLicense()
+		// model.NewTestLicense generates a E20 license
+		*l.Features.EnterprisePlugins = false
+		th.App.Srv().SetLicense(l)
+
+		// The content of the request is irrelevant. This test only cares about enterprise_plugins.
+		pRequest := &model.InstallMarketplacePluginRequest{}
+		manifest, resp := th.SystemAdminClient.InstallMarketplacePlugin(pRequest)
```

Comment:
One thing these tests are missing is that: we're not asserting if the remote server was actually called. It is entirely possible that the remote request were never made and these tests would still pass because the asserts/requires are inside the `HandlerFunc`.

One way to assert that would be to set a flag inside the `HandlerFunc` and assert that. Or make sure a plugin was successfully installed from the remote server. Either one.

Reply:
Good point :+1: Added a check to ensure the server is hit.